### ISSUE TITLE
Testssuite fails to run: pytest.PytestDeprecationWarning: The hookimpl CovPlugin.pytest_configure_node uses old-style configuration options

### ISSUE THREAD

**Issue by yurivict** [spyder-ide/qtpy#454]
```
===>   py39-QtPy-2.4.0 depends on file: /usr/local/bin/python3.9 - found
cd /usr/ports/devel/py-QtPy/work-py39/QtPy-2.4.0 && /usr/bin/env XDG_DATA_HOME=/usr/ports/devel/py-QtPy/work-py39  XDG_CONFIG_HOME=/usr/ports/devel/py-QtPy/work-py39  XDG_CACHE_HOME=/usr/ports/devel/py-QtPy/work-py39/.cache  HOME=/usr/ports/devel/py-QtPy/work-py39 PATH=/usr/local/libexec/ccache:/usr/ports/devel/py-QtPy/work-py39/.bin:/home/yuri/bin:/sbin:/bin:/usr/sbin:/usr/bin:/usr/local/sbin:/usr/local/bin PKG_CONFIG_LIBDIR=/usr/ports/devel/py-QtPy/work-py39/.pkgconfig:/usr/local/libdata/pkgconfig:/usr/local/share/pkgconfig:/usr/libdata/pkgconfig MK_DEBUG_FILES=no MK_KERNEL_SYMBOLS=no SHELL=/bin/sh NO_LINT=YES PREFIX=/usr/local  LOCALBASE=/usr/local  CC="cc" CFLAGS="-O2 -pipe  -fstack-protector-strong -fno-strict-aliasing "  CPP="cpp" CPPFLAGS=""  LDFLAGS=" -fstack-protector-strong " LIBS=""  CXX="c++" CXXFLAGS="-O2 -pipe -fstack-protector-strong -fno-strict-aliasing  "  MANPREFIX="/usr/local" CCACHE_DIR="/tmp/.ccache" BSD_INSTALL_PROGRAM="install  -s -m 555"  BSD_INSTALL_LIB="install  -s -m 0644"  BSD_INSTALL_SCRIPT="install  -m 555"  BSD_INSTALL_DATA="install  -m 0644"  BSD_INSTALL_MAN="install  -m 444" /usr/local/bin/python3.9 -m pytest -k '' -rs -v -o addopts= 
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/local/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/usr/local/lib/python3.9/site-packages/pytest/__main__.py", line 5, in <module>
    raise SystemExit(pytest.console_main())
  File "/usr/local/lib/python3.9/site-packages/_pytest/config/__init__.py", line 189, in console_main
    code = main()
  File "/usr/local/lib/python3.9/site-packages/_pytest/config/__init__.py", line 147, in main
    config = _prepareconfig(args, plugins)
  File "/usr/local/lib/python3.9/site-packages/_pytest/config/__init__.py", line 328, in _prepareconfig
    config = pluginmanager.hook.pytest_cmdline_parse(
  File "/usr/local/lib/python3.9/site-packages/pluggy/_hooks.py", line 433, in __call__
    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
  File "/usr/local/lib/python3.9/site-packages/pluggy/_manager.py", line 112, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/usr/local/lib/python3.9/site-packages/pluggy/_callers.py", line 133, in _multicall
    teardown[0].send(outcome)
  File "/usr/local/lib/python3.9/site-packages/_pytest/helpconfig.py", line 103, in pytest_cmdline_parse
    config: Config = outcome.get_result()
  File "/usr/local/lib/python3.9/site-packages/pluggy/_result.py", line 108, in get_result
    raise exc.with_traceback(exc.__traceback__)
  File "/usr/local/lib/python3.9/site-packages/pluggy/_callers.py", line 80, in _multicall
    res = hook_impl.function(*args)
  File "/usr/local/lib/python3.9/site-packages/_pytest/config/__init__.py", line 1075, in pytest_cmdline_parse
    self.parse(args)
  File "/usr/local/lib/python3.9/site-packages/_pytest/config/__init__.py", line 1425, in parse
    self._preparse(args, addopts=addopts)
  File "/usr/local/lib/python3.9/site-packages/_pytest/config/__init__.py", line 1327, in _preparse
    self.hook.pytest_load_initial_conftests(
  File "/usr/local/lib/python3.9/site-packages/pluggy/_hooks.py", line 433, in __call__
    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
  File "/usr/local/lib/python3.9/site-packages/pluggy/_manager.py", line 112, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/usr/local/lib/python3.9/site-packages/pluggy/_callers.py", line 155, in _multicall
    return outcome.get_result()
  File "/usr/local/lib/python3.9/site-packages/pluggy/_result.py", line 108, in get_result
    raise exc.with_traceback(exc.__traceback__)
  File "/usr/local/lib/python3.9/site-packages/pluggy/_callers.py", line 80, in _multicall
    res = hook_impl.function(*args)
  File "/usr/local/lib/python3.9/site-packages/pytest_enabler/__init__.py", line 50, in pytest_load_initial_conftests
    _pytest_cov_check(enabled, early_config, parser, args)
  File "/usr/local/lib/python3.9/site-packages/pytest_enabler/__init__.py", line 90, in _pytest_cov_check
    pytest_cov.plugin.pytest_load_initial_conftests(early_config, parser, args)
  File "/usr/local/lib/python3.9/site-packages/pytest_cov/plugin.py", line 119, in pytest_load_initial_conftests
    early_config.pluginmanager.register(plugin, '_cov')
  File "/usr/local/lib/python3.9/site-packages/_pytest/config/__init__.py", line 488, in register
    ret: Optional[str] = super().register(plugin, name)
  File "/usr/local/lib/python3.9/site-packages/pluggy/_manager.py", line 146, in register
    hookimpl_opts = self.parse_hookimpl_opts(plugin, name)
  File "/usr/local/lib/python3.9/site-packages/_pytest/config/__init__.py", line 459, in parse_hookimpl_opts
    return _get_legacy_hook_marks(
  File "/usr/local/lib/python3.9/site-packages/_pytest/config/__init__.py", line 373, in _get_legacy_hook_marks
    warn_explicit_for(cast(FunctionType, method), message)
  File "/usr/local/lib/python3.9/site-packages/_pytest/warning_types.py", line 170, in warn_explicit_for
    raise type(w)(f"{w}\n at {filename}:{lineno}") from None
pytest.PytestDeprecationWarning: The hookimpl CovPlugin.pytest_configure_node uses old-style configuration options (marks or attributes).
Please use the pytest.hookimpl(optionalhook=True) decorator instead
 to configure the hooks.
 See https://docs.pytest.org/en/latest/deprecations.html#configuring-hook-specs-impls-using-markers
 at /usr/local/lib/python3.9/site-packages/pytest_cov/plugin.py:226
*** Error code 1
```

py39-pytest-7.4.0
python-3.9
FreeBSD 13.2

**Comment by dalthviz** [comment]
Hi @yurivict thank you for the feedback! Checking the traceback, seems like the warning comes from `pytest-cov` Are you using the latest version (4.1.0)? Also, seems like a related issue over `pytest-cov` is closed and its fix is available starting with version 4.0.0: https://github.com/pytest-dev/pytest-cov/issues/561

**Comment by yurivict** [comment]
No, we still use pytest-cov-2.x. Upgrade is requested and is in progress.

Closing then.
Thank you for your attention!
